Runtime/CSharp/Bitcraft.StateMachine/StateManager.cs

```diff
@@ -75,26 +75,28 @@ public void SetInitialState(StateToken initialState, object data)
             RaiseOnExitEvent(null, null);
 
             CurrentState = null;
-            PerformTransitionTo(initialState, data);
+            PerformTransitionTo(null, initialState, data);
         }
 
-        private void PerformTransitionTo(StateToken stateToken, object data)
+        private void PerformTransitionTo(ActionToken actionToken, StateToken stateToken, object data)
         {
+            ActionToken triggeringActionToken = actionToken;
             StateToken targetStateToken = stateToken;
             object targetData = data;
 
             while (true)
             {
-                TransitionInfo transition = TransitionTo(targetStateToken, targetData);
+                TransitionInfo transition = TransitionTo(triggeringActionToken, targetStateToken, targetData);
                 if (transition.TargetStateToken == null)
                     break;
 
+                triggeringActionToken = transition.TriggeringAction;
                 targetStateToken = transition.TargetStateToken;
                 targetData = transition.TargetStateData;
             }
         }
 
-        private TransitionInfo TransitionTo(StateToken stateToken, object data)
+        private TransitionInfo TransitionTo(ActionToken actionToken, StateToken stateToken, object data)
         {
```

Comment:
If we add `StateManager`'s special action tokens for Init and Finalize, then we should not forget to add
```cs
if (actionToken == null)
    throw new ArgumentNullException(nameof(actionToken));
```